### PR TITLE
1906june10java batch's new content features

### PR DESCRIPTION
**Pull request by tmilesfowler**
- QC users can approve or deny content
- In the search pages, content can now be filtered by whether it is still pending approval from a QC user or has been approved. The QC user can additionally see content that has been denied.

**Important Scripts**
- SearchServiceImpl, SearchController
- ContentServiceImpl, ContentController

### CODE REVIEW

File: src/main/java/com/revature/controllers/SearchController.java [hunk 1]
```diff
@@ -33,6 +33,9 @@
 	public Set<Content> filter(@RequestBody Map<String, Object> body) {
 		@SuppressWarnings("unchecked")
```
Comment (ryanfradj) [inline src/main/java/com/revature/controllers/SearchController.java:34]: do they need to be serialized?

File: src/main/java/com/revature/repositories/ContentRepository.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
 	
 	Set<Content> findById(int id);
 	
+	Set<Content> findByStatus(String status);
+	
 	// return set of contents here; we will iterate through that set in timegraph service impl.
```
Comment (ryanfradj) [inline src/main/java/com/revature/repositories/ContentRepository.java]: delete the comment from previous

File: src/main/java/com/revature/entities/User.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+	
+	@CreationTimestamp
+	@Temporal(TemporalType.DATE)
+	@Column(name="createdDate")
+	private Date createdDate;
+	
+	@Column(name = "resetToken")
+	private String resetToken;
+	
+//	columnDefinition forces the role to be either MANAGER, CUSTOMER, or VENDOR
```
Comment (ryanfradj) [inline src/main/java/com/revature/entities/User.java]: delete comments

File: src/main/java/com/revature/entities/User.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+public User(String email, String password) {
+	super();
+	this.email = email;
+	this.password = password;
+}
+
+
+//public User(String resetToken) {
```
Comment (ryanfradj) [inline src/main/java/com/revature/entities/User.java]: delete unused block

File: src/main/java/com/revature/services/SearchServiceImpl.java [hunk 5]
```diff
@@ -153,6 +185,8 @@
 			}
 		}
 		
+		System.out.println("content statuses" + contents);
```
Comment (ryanfradj) [inline src/main/java/com/revature/services/SearchServiceImpl.java]: delete Sysout

File: src/test/java/com/revature/servicestests/SearchServiceTest.java [hunk 6]
```diff
@@ -241,10 +241,11 @@ void searchServiceOverallFilter()
 		String title = content.getTitle();
 		String format = content.getFormat();
 		List<Integer> mlist = new ArrayList<Integer>();
+		List<String> tempArr = new ArrayList<String>();
 		mlist.add(module1.getId());
 		mlist.add(module2.getId());
-
-		Set<Content> filtered = ss.filter(title, format, mlist);
+		//tempArr IS TEMPORARY TO SILENCE ERRORS
```
Comment (ryanfradj) [inline src/test/java/com/revature/servicestests/SearchServiceTest.java:268]: delete comment

File: src/test/java/com/revature/servicestests/SearchServiceTest.java [hunk 7]
```diff
@@ -324,12 +327,14 @@ void searchServiceBadFormat()
...
 		String format = content.getFormat();
 		List<Integer> mlist = new ArrayList<Integer>();
+		List<String> tempArr = new ArrayList<String>();
 		mlist.add(module1.getId());
 		mlist.add(module2.getId());
 
 		String badformat = "Document";
-		Set<Content> filtered = ss.filter(title, format, mlist);
-		filtered = ss.filter(title, badformat, mlist);
+		//tempArr IS TEMPORARY TO SILENCE ERRORS
```
Comment (ryanfradj) [inline src/test/java/com/revature/servicestests/SearchServiceTest.java:365]: delete comment

File: src/test/java/com/revature/servicestests/SearchServiceTest.java [hunk 8]
```diff
@@ -368,8 +373,10 @@ void searchServiceBadModule()
 		String format = content.getFormat();
 
 		List<Integer> mlist = new ArrayList<Integer>();
+		List<String> tempArr = new ArrayList<String>();
 		mlist.add(module3.getId());
-		Set<Content >filtered = ss.filter(title, format, mlist);
+		//temparr IS TEMPORARY TO SILENCE ERRORS
```
Comment (ryanfradj) [inline src/test/java/com/revature/servicestests/SearchServiceTest.java:412]: delete comment

**Review by ryanfradj — COMMENTED** [review]
some comments from the previous batch and one sysout

File: src/main/java/com/revature/controllers/SearchController.java (hunk 1 above)
Comment (tmilesfowler) [reply src/main/java/com/revature/controllers/SearchController.java:34]: Needed to erase warnings

File: src/main/java/com/revature/repositories/ContentRepository.java (hunk 2 above)
Comment (tmilesfowler) [reply src/main/java/com/revature/repositories/ContentRepository.java]: Added javadoc comments to all methods

File: src/main/java/com/revature/entities/User.java (hunk 3 above)
Comment (tmilesfowler) [reply src/main/java/com/revature/entities/User.java]: Deleted uneccessary contents

File: src/main/java/com/revature/entities/User.java (hunk 4 above)
Comment (tmilesfowler) [reply src/main/java/com/revature/entities/User.java]: Block deleted

File: src/main/java/com/revature/services/SearchServiceImpl.java (hunk 5 above)
Comment (tmilesfowler) [reply src/main/java/com/revature/services/SearchServiceImpl.java]: Deleted sysout

File: src/test/java/com/revature/servicestests/SearchServiceTest.java (hunk 6 above)
Comment (tmilesfowler) [reply src/test/java/com/revature/servicestests/SearchServiceTest.java:268]: Comment necessary to indicate purpose of temporary variable

File: src/test/java/com/revature/servicestests/SearchServiceTest.java (hunk 7 above)
Comment (tmilesfowler) [reply src/test/java/com/revature/servicestests/SearchServiceTest.java:365]: Comment necessary to indicate purpose of temporary variable

File: src/test/java/com/revature/servicestests/SearchServiceTest.java (hunk 8 above)
Comment (tmilesfowler) [reply src/test/java/com/revature/servicestests/SearchServiceTest.java:412]: Comment necessary to indicate purpose of temporary variable

File: src/main/java/com/revature/controllers/SearchController.java [hunk 9]
```diff
@@ -33,6 +33,7 @@
...
 		List<Integer> lst = (ArrayList<Integer>) body.get("modules");
-		return searchService.filter(body.get("title").toString(), body.get("format").toString(), lst);
+		System.out.println(body.toString());
```
Comment (Ashujauhari) [inline src/main/java/com/revature/controllers/SearchController.java:38]: Remove sysout

File: src/main/java/com/revature/services/SearchServiceImpl.java [hunk 10]
```diff
@@ -153,6 +153,22 @@
...
+			contents = cr.findByStatus(status);
+		}
+		
+		copy = new HashSet<Content>(contents);
+		
+		if(!status.equals("All")) {
+			for(Content c : copy) {
+				if (c.getStatus().equals("pending")) {
+					contents.remove(c);
+					//fakeContent = c;
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/SearchServiceImpl.java]: Remove comment

File: src/main/java/com/revature/services/SearchServiceImpl.java [hunk 11]
```diff
@@ -153,6 +153,22 @@
...
+					System.out.println("content check" + contents);
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/SearchServiceImpl.java]: Remove sysout

File: src/main/java/com/revature/entities/Content.java [hunk 12]
```diff
@@ -43,10 +48,33 @@
...
+	    
+	@Column(name = "status")
+	private String status;
+	
 	public Content() {
 		super();
 	}
 
+	public Content(int id, String title, String format, String description, String url, long dateCreated,
+            long lastModified, Set<Link> links, User userid, String status) {
```
Comment (Ashujauhari) [inline src/main/java/com/revature/entities/Content.java:64]: Update userid as per java coding standard

File: src/test/java/com/revature/servicestests/SearchServiceTest.java [hunk 13]
```diff
@@ -79,11 +80,15 @@ void searchServiceTitleTest()
 		Module module2 = new Module(0, "SECOND TEST MODULE", 0, null);
 		Module module3 = new Module(0, "THIRD TEST MODULE", 0, null);
 		
-		Content content = new Content(0, "FIRST TEST CONTENT", "Code", "FIRST TEST CONTENT DESCRIPTION", "http://www.elmo.test", new HashSet<Link>(), 1563378565, 1563378565);
+		User user = new User("TEST", "USER");
+		Content content = new Content(0, "FIRST TEST CONTENT", "Code", "FIRST TEST CONTENT DESCRIPTION", "http://www.elmo.test", 1563378565, 1563378565, 
+				new HashSet<Link>(), user, "PENDING");
 
 		module1 =mr.save(module1);
```
Comment (Ashujauhari) [inline src/test/java/com/revature/servicestests/SearchServiceTest.java:87]: Update mr as per java coding standard

File: src/test/java/com/revature/servicestests/SearchServiceTest.java [hunk 14]
```diff
@@ -241,10 +262,11 @@ void searchServiceOverallFilter()
 		String title = content.getTitle();
 		String format = content.getFormat();
 		List<Integer> mlist = new ArrayList<Integer>();
+		List<String> tempArr = new ArrayList<String>();
 		mlist.add(module1.getId());
 		mlist.add(module2.getId());
-
-		Set<Content> filtered = ss.filter(title, format, mlist);
+		//tempArr IS TEMPORARY TO SILENCE ERRORS 
+		Set<Content> filtered = ss.filter(title, format, mlist, tempArr);
```
Comment (Ashujauhari) [inline src/test/java/com/revature/servicestests/SearchServiceTest.java:269]: Update ss as per java coding standard

File: src/main/java/com/revature/services/ContentServiceImpl.java [hunk 15]
```diff
@@ -31,6 +31,8 @@
 	LinkRepository lr;
 	@Autowired
 	ModuleRepository mr;
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/ContentServiceImpl.java:33]: Update mr as per java coding standard

File: src/main/java/com/revature/services/ContentServiceImpl.java [hunk 16]
```diff
@@ -31,6 +31,8 @@
...
+	@Autowired
+	ContentService cs;
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/ContentServiceImpl.java:35]: Update cs as per java coding standard

File: src/main/java/com/revature/services/ContentServiceImpl.java [hunk 17]
```diff
@@ -185,5 +187,24 @@ public Content updateContent(Content newContent) {
...
+		if(decision.equals("true")) {
+			updated.setStatus("APPROVED");
+			cr.save(updated);
+		}
+		else if(decision.equals("false")) {
+			updated.setStatus("DENIED");
+			cr.save(updated);
+		}
+		
+		System.out.println(updated);
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/ContentServiceImpl.java:206]: Remove System.out.println

File: src/main/java/com/revature/services/SearchServiceImpl.java [hunk 18]
```diff
@@ -96,20 +96,52 @@
...
-			contents = cr.findByFormat(format);
+			if(contents == null) {
+			
+				contents = contentRepo.findByFormat(format);
+			
+			} else {
+				
+				copy = new HashSet<Content>(contents);
+				
+				for(Content c : copy) {
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/SearchServiceImpl.java:129]: Change c to content

File: src/main/java/com/revature/services/ContentServiceImpl.java [hunk 19]
```diff
@@ -31,6 +31,8 @@
 	LinkRepository lr;
```
Comment (Ashujauhari) [inline src/main/java/com/revature/services/ContentServiceImpl.java:31]: Update lr as per java coding standard

**Review by Ashujauhari — APPROVED** [review]

**Pull request merged**